### ISSUE TITLE
GPU memory allocation in for loop when using the FourierOperator

### ISSUE THREAD

**Issue by koflera** [PTB-MR/mrpro#747]
running this minimal example causes a CUDA out-of-memory error at some point. Notice that the optimizer is not even called here and that everything is within the `with torch.no_grad():`, so the error comes already by the repeatedly applying the adjoint of the fourier-operator.
Training

```
import torch
import mrpro

n_k0 = ny = nx = 160
n_k1 = 160
acceleration_factor = 4
k1_idx = torch.randperm(n_k0)[: int(n_k1 // acceleration_factor)]

traj = mrpro.data.traj_calculators.KTrajectoryCartesian()(
    n_k0=n_k0,
    k0_center=n_k0 // 2,
    k1_idx=k1_idx[:, None],
    k1_center=n_k1 // 2,
    k2_idx=torch.tensor(0),
    k2_center=0,
)
im_shape = mrpro.data.SpatialDimension(z=1, y=ny, x=nx)
k_shape = mrpro.data.SpatialDimension(z=1, y=n_k1, x=n_k0)

fourier_operator = mrpro.operators.FourierOp(
    traj=traj,
    recon_matrix=im_shape,
    encoding_matrix=k_shape,
)


class Network(torch.nn.Module):
    def __init__(
        self,
    ):
        super().__init__()
        self.parameter = torch.nn.Parameter(torch.tensor([1.0]), requires_grad=True)

    def forward(self, forward_operator: mrpro.operators.LinearOperator, kdata: torch.Tensor) -> torch.Tensor:
        (solution,) = forward_operator.H(kdata * self.parameter)

        return solution


network = Network()
optimizer = torch.optim.Adam(network.parameters(), lr=1e-4)

image = torch.randn(1, 1, 1, 160, 160, dtype=torch.complex64)
(kdata,) = fourier_operator(image)

if torch.cuda.is_available():
    network = network.cuda()
    image = image.cuda()
    kdata = kdata.cuda()
    fourier_operator = fourier_operator.cuda()

for _ in range(100000):
    print(_)
    # optimizer.zero_grad()
    with torch.no_grad():
        #     reconstruction = network(fourier_operator, kdata)
        (reconstruction,) = fourier_operator.H(kdata)

    # loss = torch.nn.functional.mse_loss(torch.view_as_real(reconstruction), torch.view_as_real(image))

    # loss.backward()
    # optimizer.step()
    # print(loss.item())


```

**Comment by koflera** [comment]
hmmm, and I just tested whether or not this happens with other operators as well. For example, with the gradient operator, everything is fine....

**Comment by koflera** [comment]
and, even more interesting, if I change the flag `adjoint_as_backward` in the FourierOperator to `False`, the problem also does not occur... any idea what this is related to @fzimmermann89?

**Comment by ckolbPTB** [comment]
By coincidence I received an email from Noah today, that his MRpro recon (CPU, no DL stuff) has the same problem. The fix with setting adjoint_as_backward to False also worked for him. I sent him an updated code last week but we also made the reconstruction problem more memory intenstive. Not sure if the problem was not there before or if he simply did not notice it.
